include/arch/riscv/arch.h

```diff
@@ -90,7 +91,7 @@ extern "C" {
  * SOC-specific function to get the IRQ number generating the interrupt.
  * __soc_get_irq returns a bitfield of pending IRQs.
  */
-extern u32_t __soc_get_irq(void);
```

Comment:
I think you need to rebase your series, `u32_t` was removed by @galak some time ago

Reply:
Thanks for pointing that out. I've rebased the code and replaced all the instances of `u32_t` that we introduced with `uint32_t`.